src/scalevoid.h

```diff
@@ -18,52 +58,122 @@ void checkWeight() {
         return;
     }
 
-    if (currentMillisScale - previousMillisScale >= intervalWeight) {
-        previousMillisScale = currentMillisScale;
+    // check for new data/start next conversion:
+    if (LoadCell.update()) newDataReady = true;
+    #if SINGLE_HX711 == 0
+        // weirdly, the library examples do not check for updates on the second cell before getting the values...
+        LoadCell2.update();
+    #endif
 
-        // check for new data/start next conversion:
-        if (LoadCell.update()) {
-            newDataReady = true;
-        }
+    if (newDataReady) {
+        if (currentMillisScale - previousMillisScale >= intervalWeight) {
+            previousMillisScale = currentMillisScale;
+            newDataReady = false;
+            w1 = LoadCell.getData();
 
-        // get smoothed value from the dataset:
-        if (newDataReady) {
-            weight = LoadCell.getData();
-            newDataReady = 0;
+            #if SINGLE_HX711 == 0
+            w2 = LoadCell2.getData();
+            // debugPrintf("Current weight: %.2f from %.2f and %.2f\n", weight, w1, w2);
+            #endif
         }
     }
+
+    #if SINGLE_HX711 == 0
+    weight = w1 + w2;
+    #else 
+    weight = w1;
+    #endif
+
+    if (calibrationON) {
+        calibrate(LoadCell, PIN_HXDAT, STO_ITEM_SCALE_CALIBRATION_FACTOR, &scaleCalibration);
+        #if SINGLE_HX711 == 0
+        calibrate(LoadCell2, PIN_HXDAT2, STO_ITEM_SCALE2_CALIBRATION_FACTOR, &scale2Calibration);
+        #endif
+        calibrationON = 0;
+    }
+
+    if (tareON) {
+        tareON = 0;
+        u8g2.clearBuffer();
+        u8g2.drawStr(0, 2, "Taring scale,");
+        u8g2.drawStr(0, 12, "remove any load!");
+        u8g2.drawStr(0, 22, "....");
+        delay(2000);
+        u8g2.sendBuffer();
+        LoadCell.tare();
+        LoadCell.setCalFactor(scaleCalibration);
+        #if SINGLE_HX711 == 0
+        LoadCell2.setCalFactor(scale2Calibration);
+        LoadCell2.tare();
+        #endif
+        u8g2.drawStr(0, 32, "done");
+        u8g2.sendBuffer();
+        delay(2000);
+    }
+    
 }
 
-/**
- * @brief Initialize scale
- */
 void initScale() {
+    boolean shouldCalibrate = calibrationON;
+
     LoadCell.begin();
-    long stabilizingtime = 2000; // tare preciscion can be improved by adding a few seconds of stabilizing time
+    #if SINGLE_HX711 == 0
+    LoadCell2.begin();
+    #endif 
+    // u8g2.clearBuffer();
```

Comment:
Why is this comented out?

Reply:
Because it's unnecessary and adds a 2 second delay to the machine startup. I'll delete it